src/ray/object_manager/object_buffer_pool.h

```diff
@@ -119,34 +122,45 @@ class ObjectBufferPool {
   /// \param chunk_index The index of the chunk.
   /// \param data The data to write into the chunk.
   void WriteChunk(const ObjectID &object_id, uint64_t chunk_index,
-                  const std::string &data);
+                  const std::string &data) ABSL_LOCKS_EXCLUDED(pool_mutex_);
 
   /// Free a list of objects from object store.
   ///
   /// \param object_ids the The list of ObjectIDs to be deleted.
   /// \return Void.
-  void FreeObjects(const std::vector<ObjectID> &object_ids);
+  void FreeObjects(const std::vector<ObjectID> &object_ids)
+      ABSL_LOCKS_EXCLUDED(pool_mutex_);
 
   /// Abort the create operation associated with an object. This destroys the buffer
   /// state, including create operations in progress for all chunks of the object.
-  void AbortCreate(const ObjectID &object_id);
+  void AbortCreate(const ObjectID &object_id) ABSL_LOCKS_EXCLUDED(pool_mutex_);
 
   /// Returns debug string for class.
   ///
   /// \return string.
-  std::string DebugString() const;
+  std::string DebugString() const ABSL_LOCKS_EXCLUDED(pool_mutex_);
 
  private:
   /// Splits an object into ceil(data_size/chunk_size) chunks, which will
   /// either be read or written to in parallel.
   std::vector<ChunkInfo> BuildChunks(const ObjectID &object_id, uint8_t *data,
                                      uint64_t data_size,
-                                     std::shared_ptr<Buffer> buffer_ref);
+                                     std::shared_ptr<Buffer> buffer_ref)
+      ABSL_EXCLUSIVE_LOCKS_REQUIRED(pool_mutex_);
+
+  /// Ensures buffer for the object exists, and creates the buffer if needed.
+  /// Returns OK if buffer exists.
+  /// Must hold pool_mutex_ when calling this function. pool_mutex_ can be released
+  /// during the call.
+  ray::Status EnsureBufferExists(const ObjectID &object_id,
+                                 const rpc::Address &owner_address, uint64_t data_size,
+                                 uint64_t metadata_size, uint64_t chunk_index)
+      ABSL_EXCLUSIVE_LOCKS_REQUIRED(pool_mutex_);
```

Comment:
Btw I think we are omitting the `ABSL_` prefix on annotations.

Reply:
Good to know, updated.